frontend/components/organiser/EventDrilldownDetailsPage.tsx

```diff
@@ -79,13 +81,13 @@ const EventDrilldownDetailsPage = ({
     setEditTitle(false);
   };
 
-  const [editDate, setEditDate] = useState(false);
-  const [newEditDate, setNewEditDate] = useState("");
-  const [date, setDate] = useState("");
+  const [editStartDate, setEditStartDate] = useState(false);
+  const [newEditStartDate, setNewEditStartDate] = useState("");
+  const [startDate, setStartDate] = useState("");
 
-  const [editTime, setEditTime] = useState(false);
-  const [newEditTime, setNewEditTime] = useState("");
-  const [time, setTime] = useState("");
+  const [editStartTime, setEditStartTime] = useState(false);
+  const [newEditStartTime, setNewEditStartTime] = useState("");
+  const [startTime, setStartTime] = useState("");
```

Comment:
can we not just slap the start and endtime straight into the useState? why do we have to do it in the useEffect?

im also a bit concerned at the amount of states being instantiated here.. its really hard to read, maybe we can discuss alternatives tmrw?

Reply:
uh correct me if im wrong but i had these states so that if the user reverts changes like they make changes then click the x button, id need to hold the value it had previously so thats why theres neweditstartdate and startdate, then obviously whether it is editable is another state

Reply:
but yeah it is very hard to read but im not too sure what an elegant solution would be

Reply:
Apart from holding all the values in an object, I'm not really sure of a cleaner way to achieve this.

Grouping the states line by line is a good start which youve already done.

Reply:
nah there is a way owen outlined to me last night but out of scope - can revisit later

Reply:
Can you put a TODO comment for this and make a JIRA issue.